src/k8s/pkg/k8sd/api/component.go

```diff
@@ -139,3 +144,21 @@ func ingressComponentPut(s *state.State, r *http.Request) response.Response {
 
 	return response.SyncResponse(true, &api.UpdateIngressComponentResponse{})
 }
+
+func gatewayComponentPut(s *state.State, r *http.Request) response.Response {
+	var req api.UpdateGatewayComponentRequest
+
+	snap := snap.SnapFromContext(s.Context)
+
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.SmartError(fmt.Errorf("failed to decode request: %w", err))
+	}
+
+	if req.Status == api.ComponentEnable {
+		err = component.EnableGatewayComponent(snap)
+	} else {
+		err = component.DisableGatewayComponent(snap)
+	}
+	return response.SyncResponse(true, &api.UpdateGatewayComponentResponse{})
```

Comment:
obviously not right -- but maybe something like this?

```suggestion
    if err != nil {
       return response.SmartError(fmt.Errorf("failed to toggle gateway: %w", err))
    }
	return response.SyncResponse(true, &api.UpdateGatewayComponentResponse{})
```